oskari-geojson/src/main/java/org/oskari/geojson/GeoJSONSchemaDetector.java

```diff
@@ -0,0 +1,183 @@
+package org.oskari.geojson;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.geotools.feature.simple.SimpleFeatureTypeBuilder;
+import org.opengis.feature.simple.SimpleFeatureType;
+import org.opengis.referencing.crs.CoordinateReferenceSystem;
+
+import com.vividsolutions.jts.geom.Geometry;
+import com.vividsolutions.jts.geom.GeometryCollection;
+import com.vividsolutions.jts.geom.LineString;
+import com.vividsolutions.jts.geom.MultiLineString;
+import com.vividsolutions.jts.geom.MultiPoint;
+import com.vividsolutions.jts.geom.MultiPolygon;
+import com.vividsolutions.jts.geom.Point;
+import com.vividsolutions.jts.geom.Polygon;
+
+public class GeoJSONSchemaDetector {
+
+    @SuppressWarnings("unchecked")
+    public static SimpleFeatureType getSchema(Map<String, Object> json, CoordinateReferenceSystem crs) {
+        GeoJSONReader2.replaceMapsWithGeometries(json);
```

Comment:
This creates a side-effect by modifying the input as well as returning the SimpleFeatureType. That might cause problems later on.